From 5b2c3bb86450cd0b5611d9abad9ece7a4ca9a70e Mon Sep 17 00:00:00 2001
From: Ian Hua <ianhua@google.com>
Date: Thu, 6 Jul 2023 10:05:36 +0000
Subject: [PATCH] Fix out of Bounds Read in convertSubgraphFromHAL in
 ShimConverter.cpp in libneuralnetworks_shim_static

Bug: 269270167
Test: N/A
(cherry picked from commit 4bf7bb6b50b412678a681d29f7ced70a4d737762)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:6e1bbe89e08936a1cad152a59ae82a3773c51cca)
Merged-In: I33272284b965efcbb531f64cbf838a0d59c28e00
Change-Id: I33272284b965efcbb531f64cbf838a0d59c28e00
---
 shim_and_sl/ShimConverter.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/shim_and_sl/ShimConverter.cpp b/shim_and_sl/ShimConverter.cpp
index b0f914c3..b8619d0f 100644
--- a/shim_and_sl/ShimConverter.cpp
+++ b/shim_and_sl/ShimConverter.cpp
@@ -150,6 +150,10 @@ ANeuralNetworksModel* convertSubgraphFromHAL(
                 break;
             }
             case OperandLifeTime::CONSTANT_POOL: {
+                if (operand.location.poolIndex >= memoryPools.size()) {
+                    *errorStatus = ErrorStatus::INVALID_ARGUMENT;
+                    return nullptr;
+                }
                 resultModel.setOperandValueFromMemory(
                         i, memoryPools[operand.location.poolIndex].get(), operand.location.offset,
                         operand.location.length);
-- 
2.41.0.585.gd2178a4bd4-goog

